spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -111,6 +111,11 @@ public void setOutputChannelName(String outputChannelName) {
 		return this.outputChannel;
 	}
 
+	@Override
+	public String getComponentType() {
+		return "message-producer";
```

Comment:
I think neither of abstract class should provide an impl.
That is really responsibility of the specific one.
Without an implementation here, we may spot a bug somewhere down the hierarchy, e.g. `ReactiveMessageSourceProducer`.